src/app/catalogue-search/catalogue-search.service.ts

```diff
@@ -53,20 +54,55 @@ export class CatalogueSearchService {
     params: CatalogueSearchParameters
   ): Observable<CatalogueSearchResultSet> {
     const [page, pageParams] = this.getPageParameters(params);
-    const query: SearchQueryParameters = {
-      ...this.getCommonQueryParameters(params),
-      ...pageParams,
-      searchTerm: this.getSearchTerm(params)
-    };
 
-    return this.searchCatalogue(query, params.context).pipe(
-      map((searchResults) => {
-        return {
-          count: searchResults.count,
-          pageSize: pageParams.max!, // eslint-disable-line @typescript-eslint/no-non-null-assertion
-          page,
-          items: searchResults.items
+    // Create a stream emitting the profile filters to be used in the query
+    const profileFieldsQueryData$ = params.profileFiltersDto
+      ? this.getProfileFiltersForQuery(params.profileFiltersDto)
+      : of<ProfileFieldQueryData[]>([]);
+
+    return profileFieldsQueryData$.pipe(
+      switchMap((profileFieldsQueryData) => {
+        // Create the searchQuery with the emitted profile info
+        const query: SearchQueryParameters = {
+          ...this.getCommonQueryParameters(params),
+          ...pageParams,
+          searchTerm: this.getSearchTerm(params),
+          profileFields: profileFieldsQueryData
         };
+
+        return this.searchCatalogue(query, params.context).pipe(
+          map((searchResults) => {
+            return {
+              count: searchResults.count,
+              pageSize: pageParams.max!, // eslint-disable-line @typescript-eslint/no-non-null-assertion
+              page,
+              items: searchResults.items
+            };
+          })
+        );
+      })
+    );
+  }
+
+  private getProfileFiltersForQuery(
```

Comment:
I was expecting eslint to complain about a private function in between non private functions, but it did not. So, I guess this is all right.